src/main/java/org/apache/maven/plugins/shade/mojo/ShadeMojo.java

```diff
@@ -823,6 +844,50 @@ private File resolveArtifactSources( Artifact artifact )
             }
         }
 
+        // first check for backward compatibility this is not already configured explicitly
+        boolean addExclusion = true;
+        if ( this.filters != null )
+        {
+            for ( final ArchiveFilter filter : this.filters )
+            {
+                if ( filter.getExcludes() != null
+                        && filter.getExcludes().contains( "META-INF/*.SF" )
+                        && filter.getExcludes().contains( "META-INF/*.DSA" )
+                        && filter.getExcludes().contains( "META-INF/*.RSA" )
+                        && "*:*".equals( filter.getArtifact() ) )
+                {
+                    addExclusion = false;
+                    break;
+                }
+            }
+        }
+        if ( addExclusion )
+        {
+            getLog().debug( "Adding META-INF/*.SF, META-INF/*.DSA and META-INF/*.RSA exclusion" );
+
+            if ( artifacts == null )
+            {
+                artifacts = getArtifactIds();
+            }
+            simpleFilters.add( new SimpleFilter(
+                    getMatchingJars( artifacts , new ArtifactId( "*:*" ) ),
+                    Collections.<String>emptySet(),
+                    new HashSet<String>( Arrays.asList( "META-INF/*.SF", "META-INF/*.DSA", "META-INF/*.RSA" ) ) ) );
+        }
+
+        for ( final Filter filter : ServiceLoader.load( Filter.class ) )
```

Comment:
Using ServiceLoader requires a separate JIRA issue, and I wonder if this is going to work. See https://issues.apache.org/jira/browse/MNG-6275